### ISSUE TITLE
Bug with rights on a stored and computed field

### ISSUE THREAD

**Issue by sylvainvanhoof** [odoo/odoo#4824]
Hello, 

I've a huge problem with rights and computed field.
First, you'll find a test module to demonstrate my problem : https://owncloud.sylvainvh.be/index.php/s/QjoWl7E795yloL5

## Configuration
We use Odoo v8 with the commit 9ab18cd149fccbd844e075caf25e7239c2939dde

## What is the sitation:
I've a computed and stored field on the model 'res.partner'. This field display the name and the phone of the partner (this is a stupid example). This field can be readonly or not but it must be stored.

```
@api.multi
@api.depends('name','phone')
def _get_partner_code(self):
    for partner in self:
        partner.partner_code = '%s_%s' % (partner.name, partner.phone)

partner_code = Char(compute=_get_partner_code, store=True)
```

Moreover, there are special rules on partner. A user can create a customer BUT he cannot write on it (according a rule). This case is use for a special and big project. We can not do otherwise. 

## What is the problem:
When I want to create a new customer, an error is raised:
> Access Denied
>
> The requested operation cannot be completed due to security restrictions. Please contact your  system administrator.
> 
> (Document type: Partner, Operation: write)

## Why ?
The method create will check for computed field. For each computed field, the ORM will call the method recompute() and call the method _write() with the value of our field. However, we cannot write on a partner. 

```
@api.model
def recompute(self):
    """ Recompute stored function fields. The fields and records to
        recompute have been determined by method :meth:`modified`.
    """
    while self.env.has_todo():
        field, recs = self.env.get_todo()
        # evaluate the fields to recompute, and save them to database
        for rec, rec1 in zip(recs, recs.with_context(recompute=False)):
            try:
                values = rec._convert_to_write({
                    f.name: rec[f.name] for f in field.computed_fields
                })
                rec1._write(values)
            except MissingError:
                pass
        # mark the computed fields as done
        map(recs._recompute_done, field.computed_fields)
```

## Who fix this bug
According to me, this method should be call with the user sudo(). There is a difference between create a new record (and save all computed method) and write on a existing record.

This is a very disturbing problem. 

Thank for all,
Sylvain Van Hoof for Eezee-it

**Comment by intero-chz** [comment]
I'm on your side with this case. With CRUD rights management it's a bit strange that you have to have write rights to create records. :+1:

**Comment by rco-odoo** [comment]
@sylvainvanhoof please check the alternative fix in #5087: instead of *always* recomputing fields as user admin, we introduce an explicit flag on the field to do it.

This should provide you a solution without breaking the behavior of other modules.
